### ISSUE TITLE
Clippy not reporting lints from path dependencies

### ISSUE THREAD

**Issue by vdrn** [rust-lang/rust-clippy#11092]
### Summary

I'm not sure if this is a bug with clippy or if I'm doing something wrong.

Running clippy in a workspace with path dependencies does not seem to report any clippy specific lints (rustc lints still get reported) from those dependencies.

My workspace has this structure
```
package1/src/lib.rs
package1/Cargo.toml
src/main.rs
target/
Cargo.toml
```
Running `cargo clippy` in the root directory reports **rustc** lints for the whole workspace, but only **clippy** lints for `src`.
I've tried running `cargo clippy -p package1` but it still does not report any clippy lints.

What does work is running `cargo clippy` from `package1` dir, but that creates another target dir `package1/target`.

### Reproducer

Relevant  code (full repo https://github.com/vdrn/clippy_not_reporting_lints/):
package1/src/lib.rs
```rust
pub fn test_fn()->bool{
  let unused = true;
  let a = false;
  return *&a;
}
```
src/main.rs
```rust
use package1::test_fn;

fn main() {
    test_fn();
    println!("Hello, world!");
}
```
I've tried
```bash
cargo clippy
```
and
```bash
cargo clippy -p package1
```

I expected to see this happen:
Lints `unused_variables`,  `clippy:deref_addrof` and `clippy:needless_return` are reported

Instead, this happened:
Only `unused_variables` is reported.

### Version

```text
rustc 1.72.0-nightly (36fb58e43 2023-06-26)
binary: rustc
commit-hash: 36fb58e433c782e27dd41034284e157cf86d587f
commit-date: 2023-06-26
host: x86_64-unknown-linux-gnu
release: 1.72.0-nightly
LLVM version: 16.0.5
```

**Comment by Alexendoo** [comment]
Cargo will implicitly add path dependencies to the workspace, but only if you have `[workspace]` in your `Cargo.toml`. If you add that to the end of the root `Cargo.toml` it should pick it up as a workspace member

**Comment by vdrn** [comment]
Thanks, solves my issue.
